File: db/java/app/src/main/java/com/example/sales/domain/model/Supplier.java
package com.example.sales.domain.model;

import java.time.LocalDateTime;

/**
 * 仕入先マスタのEntityクラス
 */
public class Supplier {
    private String supplierCode;
    private Integer supplierBranch;
    private String supplierName;
    private String supplierNameKana;
    private String supplierUserName;
    private String supplierDepartmentName;
    private String supplierZipCode;
    private String supplierState;
    private String supplierAddress1;
    private String supplierAddress2;
    private String supplierTel;
    private String supplierFax;
    private String supplierEmail;
    private Integer supplierCloseDate;
    private Integer supplierPayMonths;
    private Integer supplierPayDates;
    private Integer supplierPayMethod;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // デフォルトコンストラクタ
    public Supplier() {}

    // Getter/Setter
    public String getSupplierCode() { return supplierCode; }
    public void setSupplierCode(String supplierCode) { this.supplierCode = supplierCode; }

    public Integer getSupplierBranch() { return supplierBranch; }
    public void setSupplierBranch(Integer supplierBranch) { this.supplierBranch = supplierBranch; }

    public String getSupplierName() { return supplierName; }
    public void setSupplierName(String supplierName) { this.supplierName = supplierName; }

    public String getSupplierNameKana() { return supplierNameKana; }
    public void setSupplierNameKana(String supplierNameKana) {
        this.supplierNameKana = supplierNameKana;
    }

    public String getSupplierUserName() { return supplierUserName; }
    public void setSupplierUserName(String supplierUserName) {
        this.supplierUserName = supplierUserName;
    }

    public String getSupplierDepartmentName() { return supplierDepartmentName; }
    public void setSupplierDepartmentName(String supplierDepartmentName) {
        this.supplierDepartmentName = supplierDepartmentName;
    }

    public String getSupplierZipCode() { return supplierZipCode; }
    public void setSupplierZipCode(String supplierZipCode) {
        this.supplierZipCode = supplierZipCode;
    }

    public String getSupplierState() { return supplierState; }
    public void setSupplierState(String supplierState) { this.supplierState = supplierState; }

    public String getSupplierAddress1() { return supplierAddress1; }
    public void setSupplierAddress1(String supplierAddress1) {
        this.supplierAddress1 = supplierAddress1;
    }

    public String getSupplierAddress2() { return supplierAddress2; }
    public void setSupplierAddress2(String supplierAddress2) {
        this.supplierAddress2 = supplierAddress2;
    }

    public String getSupplierTel() { return supplierTel; }
    public void setSupplierTel(String supplierTel) { this.supplierTel = supplierTel; }

    public String getSupplierFax() { return supplierFax; }
    public void setSupplierFax(String supplierFax) { this.supplierFax = supplierFax; }

    public String getSupplierEmail() { return supplierEmail; }
    public void setSupplierEmail(String supplierEmail) { this.supplierEmail = supplierEmail; }

    public Integer getSupplierCloseDate() { return supplierCloseDate; }
    public void setSupplierCloseDate(Integer supplierCloseDate) {
        this.supplierCloseDate = supplierCloseDate;
    }

    public Integer getSupplierPayMonths() { return supplierPayMonths; }
    public void setSupplierPayMonths(Integer supplierPayMonths) {
        this.supplierPayMonths = supplierPayMonths;
    }

    public Integer getSupplierPayDates() { return supplierPayDates; }
    public void setSupplierPayDates(Integer supplierPayDates) {
        this.supplierPayDates = supplierPayDates;
    }

    public Integer getSupplierPayMethod() { return supplierPayMethod; }
    public void setSupplierPayMethod(Integer supplierPayMethod) {
        this.supplierPayMethod = supplierPayMethod;
    }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) { this.createdBy = createdBy; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public String getUpdatedBy() { return updatedBy; }
    public void setUpdatedBy(String updatedBy) { this.updatedBy = updatedBy; }
}

File: db/go/sales_management/internal/model/supplier.go
package model

import "time"

// Supplier は仕入先マスタのドメインモデルです
// 取引先のうち、仕入先としての詳細情報を保持する
type Supplier struct {
	SupplierCode           string    `db:"仕入先コード"`
	SupplierBranch         int       `db:"仕入先枝番"`
	SupplierCategory       int       `db:"仕入先区分"`
	SupplierName           string    `db:"仕入先名"`
	SupplierNameKana       string    `db:"仕入先名カナ"`
	EmployeeCode           string    `db:"担当社員コード"`
	SupplierContactName    string    `db:"仕入先担当者名"`
	SupplierDepartmentName string    `db:"仕入先担当部署名"`
	SupplierPostalCode     string    `db:"仕入先郵便番号"`
	SupplierPrefecture     string    `db:"仕入先都道府県"`
	SupplierAddress1       string    `db:"仕入先住所１"`
	SupplierPhone          string    `db:"仕入先電話番号"`
	SupplierEmail          string    `db:"仕入先メールアドレス"`
	ClosingDay             int       `db:"仕入先締日"`
	PaymentMonth           int       `db:"仕入先支払月数"`
	PaymentDay             int       `db:"仕入先支払日"`
	PaymentMethod          int       `db:"仕入先支払方法"`
	CreatedAt              time.Time `db:"作成日時"`
	CreatedBy              string    `db:"作成者名"`
	UpdatedAt              time.Time `db:"更新日時"`
	UpdatedBy              string    `db:"更新者名"`
}
